backend/src/modules/authentication/controllers/auth/auth.controller.ts

```diff
@@ -2,27 +2,60 @@ import { Controller, Post, Body, Response, Request } from '@nestjs/common';
 import * as express from 'express';
 
 import { userIdCookieKey, userIdCookieOptions } from '../../common';
+import { UserService } from '../../services/user/user.service';
+import { UserLoginDto } from '../../models/dto/UserLogin.dto';
+import { UserSignupDto } from '../../models/dto/UserSignup.dto';
 
 @Controller('api/auth')
 export class AuthController {
-  @Post('signup')
-  signUp(@Body() body, @Request() request: express.Request) {
-    // console.log(request.signedCookies);
-
+  constructor(private readonly userService: UserService) {}
 
-    // TODO: call users service to create new user
+  @Post('signup')
+  async signUp(@Body() body: UserSignupDto, @Response() response: express.Response) {
+    const isValid = await this.userService.isValidSignupDto(body);
+    if (body && !isValid) {
+      response.status(401);
+      response.send({ status: 'error', message: 'Invalid form' });
```

Comment:
maybe after this there should be 'return;' for interrupting execution of function

Reply:
response.send interrupt function execution